Rewrite Config Documentation

mikaell opened the pull request:
This PR updates parts of Scout's documentation. Ref #2330.

Highlights:

* Alphabetical order of parameters
* Remove duplicate documentation
* Mark parameters to be deprecated
* Overview of mandatory parameters
* No parameters are removed.

northwestwitch commented on docs/admin-guide/load-config.md:
```diff
@@ -1,125 +1,92 @@
-# The load config
+# The Load Config File
```
👍 Perhaps change the index (side column, see pic) accordingly?

![image](https://user-images.githubusercontent.com/28093618/116504450-95773b00-a8b8-11eb-9b37-6db6ae22408c.png)

mikaell replied:
Fixed!

northwestwitch commented on docs/admin-guide/load-config.md:
```diff
@@ -1,125 +1,92 @@
+# The Load Config File
+
 
 Scout have the possibility to store loads of information about a case and the samples that are included. It is cumbersome to specify to many parameters on the command line so there is an option to give this information in a yaml formated config file.
```
This part has many typos, what about fixing it into something:

```
A case and its individuals (or samples, in cancer track) can be uploaded into Scout using specially formatted .yaml config files. Config files contain information about the analysis, panels used, path to VCF and eventual alignment files and more.
The following template illustrates the basic structure of the .yaml config file:
```

mikaell replied:
Done

northwestwitch commented on docs/admin-guide/load-config.md:
```diff
@@ -1,125 +1,92 @@
-
-### Minimal config
+### Depreication Warnings:
+`bam_file`, `bam_path` and `alignment_path` are redundant in internal usage. Future versions of Scout will only
```
👍

northwestwitch commented on docs/admin-guide/load-config.md:
````diff
@@ -1,125 +1,92 @@
-The basic structure of a load config looks like:
-
-
-```yaml
````
I don't think you should remove this part, because this is the template that users can copy and modify to load  their own data, It's already formatted as a yaml so it's very useful to have. Moreover it shows how the yaml syntax works.

northwestwitch commented on docs/admin-guide/load-config.md:
```diff
@@ -1,125 +1,92 @@
+### Configuration Parameters
+Below are available configuration parameters for a Scout case. Names marked with asterix (*) are mandatory. 
+
+- **analysis_date(*)** Time for analysis in datetime format. Defaults to time of uploading. Example `2016-10-12 14:00:46`
```
Nice with the alphabetical order!

northwestwitch commented on docs/admin-guide/load-config.md:
```diff
@@ -1,125 +1,92 @@
+- **gene_fusion_report_research** Path to a static file containing a gene fusion report produced by [Arriba][arriba]. Generated from research data.
+- **gene_panels** List of gene panels. This will specify what panels the case has been run with.
+- **human_genome_build** Version of genome version used.
+- **lims_id** Lims Id used.
```
```suggestion
- **lims_id** case ID in Lims
```

mikaell replied:
Fixed!

northwestwitch commented on docs/admin-guide/load-config.md:
```diff
@@ -1,125 +1,92 @@
+- **coverage_qc_report** Path to html file with coverage and qc report.
+- **default_gene_panels** List of default gene panels. Variants from the genes in the gene panels specified will be shown when opening the case in scout
+- **delivery_report**: Path to html delivery report.
+- **family(*)**  Unique id of family in the configuration.
```
```suggestion
- **family(*)**  Unique ID of the case.
```

mikaell replied:
Fixed!

northwestwitch commented on docs/admin-guide/load-config.md:
```diff
@@ -1,125 +1,92 @@
+- **rank model version** Which rank model that was used when scoring the variants.
+- **rank_score_treshold** Only include variants with a rank score above this treshold.
+- **samples** List of samples included in the case:
+	- **alignment_path** Path to bam file to view alignments
```
```suggestion
	- **alignment_path** Path to bam/cram file to view alignments
```

northwestwitch commented on docs/admin-guide/load-config.md:
```diff
@@ -1,125 +1,92 @@
+- **samples** List of samples included in the case:
+	- **alignment_path** Path to bam file to view alignments
+	- **analysis_type** specifies the analysis type for the sample
+	- **bam_file** Path to bam file to view alignments **WARNING:** Soon to be deprecated, use *alignment_path*
```
```suggestion
	- **bam_file** Path to bam/cram file to view alignments **WARNING:** Soon to be deprecated, use *alignment_path*
```

Nice with the deprecation warnings! 🥇

northwestwitch commented on docs/admin-guide/load-config.md:
```diff
@@ -1,125 +1,92 @@
+- **rank_score_treshold** Only include variants with a rank score above this treshold.
+- **samples** List of samples included in the case:
+	- **alignment_path** Path to bam file to view alignments
+	- **analysis_type** specifies the analysis type for the sample
```
```suggestion
	- **analysis_type** specifies the analysis type for the sample. Options: wgs, wes, panel
```

mikaell replied:
Done!

northwestwitch commented on docs/admin-guide/load-config.md:
```diff
@@ -1,125 +1,92 @@
+	- **alignment_path** Path to bam file to view alignments
+	- **analysis_type** specifies the analysis type for the sample
+	- **bam_file** Path to bam file to view alignments **WARNING:** Soon to be deprecated, use *alignment_path*
+	- **bam_path** Path to bam file to view alignments **WARNING:** Soon to be deprecated, use *alignment_path*
```
```suggestion
	- **bam_path** Path to bam/cram file to view alignments **WARNING:** Soon to be deprecated, use *alignment_path*
```

mikaell replied:
Done!

northwestwitch commented on docs/admin-guide/load-config.md:
```diff
@@ -1,125 +1,92 @@
+	- **sex(*)** Sex of the sample in human readable format
+	- **tiddit_coverage_wig** Path to wig file to view alignments [Reference][tiddit]
+	- **tissue_type** Sample tissue origin i.e. blood, muscle, 
+	- **tmb** Tumor mutational burden
```
```suggestion
	- **tmb** Tumor mutational burden (tumor cases only)
```

mikaell replied:
Done!

northwestwitch commented on docs/admin-guide/load-config.md:
```diff
@@ -1,125 +1,92 @@
+	- **tiddit_coverage_wig** Path to wig file to view alignments [Reference][tiddit]
+	- **tissue_type** Sample tissue origin i.e. blood, muscle, 
+	- **tmb** Tumor mutational burden
+	- **tumor_purity** Purity of tumor sample
```
```suggestion
	- **tumor_purity** Purity of tumor sample (tumor cases only)
```

mikaell replied:
Done!

northwestwitch commented on docs/admin-guide/load-config.md:
```diff
@@ -1,125 +1,92 @@
+	- **tissue_type** Sample tissue origin i.e. blood, muscle, 
+	- **tmb** Tumor mutational burden
+	- **tumor_purity** Purity of tumor sample
+	- **tumor_type** Type of tumor
```
```suggestion
	- **tumor_type** Type of tumor (tumor cases only)
```

mikaell replied:
Done!

northwestwitch commented on docs/admin-guide/load-config.md:
```diff
@@ -1,125 +1,92 @@
+- **track** Type of track: {"rare", "cancer"}. Default: "rare".
+- **vcf_file** A dictionary with vcf files
+- **sv_rank_model_version** the SV rank model version used when scoring SV variants
+- **vcf_cancer** Path to canver vcf file.
```
```suggestion
- **vcf_cancer** Path to canver vcf file (tumor cases only).
```

northwestwitch commented on docs/admin-guide/load-config.md:
```diff
@@ -1,125 +1,92 @@
+- **vcf_file** A dictionary with vcf files
+- **sv_rank_model_version** the SV rank model version used when scoring SV variants
+- **vcf_cancer** Path to canver vcf file.
+- **vcf_cancer_research** Path to vcf file with all variants.
```
```suggestion
- **vcf_cancer_research** Path to vcf file with all variants (tumor cases only).
```

northwestwitch commented on docs/admin-guide/load-config.md:
```diff
@@ -1,125 +1,92 @@
+- **synopsis** Synopsis of case.
+- **sv_rank_model_version** String
+- **track** Type of track: {"rare", "cancer"}. Default: "rare".
+- **vcf_file** A dictionary with vcf files
```
This param doesn't exist in the yaml config file
```suggestion
```

mikaell replied:
Removed!

northwestwitch commented on docs/admin-guide/load-config.md:
```diff
@@ -1,125 +1,92 @@
+- **sv_rank_model_version** the SV rank model version used when scoring SV variants
+- **vcf_cancer** Path to canver vcf file.
+- **vcf_cancer_research** Path to vcf file with all variants.
+- **vcf_snv** Path to snv vcf file.
```
```suggestion
- **vcf_snv** Path to SNV vcf file containing only clinical variants (a subset of all variants)
```

mikaell replied:
Done!

northwestwitch commented on docs/admin-guide/load-config.md:
```diff
@@ -1,125 +1,92 @@
+- **vcf_cancer** Path to canver vcf file.
+- **vcf_cancer_research** Path to vcf file with all variants.
+- **vcf_snv** Path to snv vcf file.
+- **vcf_snv_research** Path to vcf file with all variants.
```
```suggestion
- **vcf_snv_research** Path to vcf file with all SV variants.
```

northwestwitch commented on docs/admin-guide/load-config.md:
```diff
@@ -1,125 +1,92 @@
+- **vcf_cancer_research** Path to vcf file with all variants.
+- **vcf_snv** Path to snv vcf file.
+- **vcf_snv_research** Path to vcf file with all variants.
+- **vcf_sv** Path to sv vcf file
```
```suggestion
- **vcf_sv** Path to SV vcf file containing only clinical variants (a subset of all variants)
```

northwestwitch commented on docs/admin-guide/load-config.md:
```diff
@@ -1,125 +1,92 @@
+- **vcf_snv** Path to snv vcf file.
+- **vcf_snv_research** Path to vcf file with all variants.
+- **vcf_sv** Path to sv vcf file
+- **vcf_sv_research** Path to vcf file with all variants.
```
```suggestion
- **vcf_sv_research** Path to vcf file with all  SV variants
```

northwestwitch commented on docs/admin-guide/load-config.md:
```diff
@@ -148,6 +115,9 @@ By giving a path to each individual vcf2cytosure-file these are made available
 for download on the case page. Such SV files can be visualized using standard arrayCGH
```
This text here could be added to expand the field vcf2cytosure in the explanation above, and this part  could be removed

mikaell replied:
Done!

northwestwitch commented on docs/admin-guide/load-config.md:
```diff
@@ -148,6 +115,9 @@ By giving a path to each individual vcf2cytosure-file these are made available
-[peddy]: https://github.com/brentp/peddy
-[multiqc]: https://github.com/ewels/multiqc
 [arriba]: https://arriba.readthedocs.io/en/latest/
+[bjhall]: https://github.com/bjhall/upd
```
```suggestion
[upd]: https://github.com/bjhall/upd
```

northwestwitch commented on docs/admin-guide/load-config.md:
```diff
@@ -1,125 +1,92 @@
+	- **tmb** Tumor mutational burden
+	- **tumor_purity** Purity of tumor sample
+	- **tumor_type** Type of tumor
+	- **upd_regions_bed** Path to bed file to view alignments [Reference][bjhall]
```
```suggestion
	- **upd_regions_bed** Path to bed file to view alignments [Reference][upd]
```

bjhall is the name of our collaborator. upd is the name of the software.

mikaell replied:
Fixed!

northwestwitch commented on docs/admin-guide/load-config.md:
```diff
@@ -1,125 +1,92 @@
+	- **tumor_purity** Purity of tumor sample
+	- **tumor_type** Type of tumor
+	- **upd_regions_bed** Path to bed file to view alignments [Reference][bjhall]
+	- **upd_sites_bed** Path to bed file to view alignments [Reference][bjhall]
```
```suggestion
	- **upd_sites_bed** Path to bed file to view alignments [Reference][upd]
```

northwestwitch commented on docs/admin-guide/load-config.md:
```diff
@@ -148,6 +115,9 @@ By giving a path to each individual vcf2cytosure-file these are made available
+[multiqc]: https://github.com/ewels/multiqc
+[peddy]: https://github.com/brentp/peddy
+[tiddit]: https://github.com/SciLifeLab/TIDDIT
+[rhocall]: https://github.com/dnil/rhocall
```
r of rhocall comes before t of tiddit, in case you wanted then in alphabetical order.

northwestwitch reviewed (changes requested):
Nice with the alphabetical ordering of the fields: it will make it much easier to find the config keys for the users!

I have some suggestions for improvement (small things mostly). 

I've noticed that you you removed the yaml template with all config options from this page but I think you should reintroduce it, since that is the part that can be directly copied and customized. It also shows how the yaml looks like. It's not sufficient to explain what the fields are, it's very useful to see a practical example.

mikaell commented:
> I've noticed that you you removed the yaml template with all config options from this page but I think you should reintroduce it, since that is the part that can be directly copied and customized. It also shows how the yaml looks like. It's not sufficient to explain what the fields are, it's very useful to see a practical example.

I don't like keeping duplicated information. It will always end up diverging like it had done in this document. I suggest instead:
1. extend the minimal configuration provided to a copy/paste friendly example
2. mention path to the `demo/643594.config.yaml` for more example

And Yaml format should be mentioned with a reference. Think I forgot that.

northwestwitch commented on docs/admin-guide/load-config.md:
```diff
@@ -1,125 +1,92 @@
+- **gene_panels** List of gene panels. This will specify what panels the case has been run with.
+- **human_genome_build** Version of genome version used.
+- **lims_id** Lims Id used.
+- **madeline** Path to a madeline pedigree file in xml format.
```
mt-bam is missing in this list and I think also in the template. Could you add it?

It's the path to the reduced mitochondrial  bam/cram alignment file

mikaell replied:
OK! Will do.

northwestwitch commented:
> I don't like keeping duplicated information. It will always end up diverging like it had done in this document. I suggest instead:
> 
> 1. extend the minimal configuration provided to a copy/paste friendly example
> 2. mention path to the `demo/643594.config.yaml` for more example
> 
> And Yaml format should be mentioned with a reference. Think I forgot that.


It's not really duplicated information, one part is how the file looks like, with a link to copy it to clipboard (top right corner)

![image](https://user-images.githubusercontent.com/28093618/116509530-c610a200-a8c3-11eb-842f-440b69854b07.png)

And the other part is the explanation of the keys. I think they complement each other in the same page. Otherwise you have to download the file from the demo and cross check it with this page in the guide, which is a bit cumbersome.. What do you think @moonso @dnil ?

northwestwitch commented:
Adding a link to the demo config file is a good idea!

mikaell commented:
I've now added types to the configuration description (forgot that earlier). The format is:
`<Parmeter name>: <Type> <Description and if applicable set/range of accepted values>`


Looks like this in my quicklook:

<img width="752" alt="image" src="https://user-images.githubusercontent.com/1150065/116690759-25e97480-a9ba-11eb-98ea-c048d8c74c95.png">



So, @dnil and @moonso  -what do you say in the matter of keeping "The basic structure of a load config looks like"

dnil commented:
> So, @dnil and @moonso -what do you say in the matter of keeping "The basic structure of a load config looks like"

Sorry, I didn't really follow the discussion, but I sort of see both sides. Keeping untested things like this in docs, rather than say in the code makes me uneasy regardless if done as copy-pasteable or template form. In-code docs and rendering is also not perfect, but better. To be pragamatic, could the example load config from the demo dir be embedded in a `<code>` or such perhaps? Or at least linked with a permalink/master relative link?

moonso commented:
Hi, avoiding to duplicate information is always good in the long run. I think it is fine to point at examples. Also I would consider to point at the pydantic model in some way from the docs as it is a very nice way to expose config structures etc.

northwestwitch commented:
> pydantic model in some way from the docs as it is a very nice way to expose config structures etc.

Good idea but that PR is not merged yet. Perhaps later!

northwestwitch commented:
Why killed? I think it's good, and I think it is fine to use the link to the demo yaml file for reference but I was waiting for you to add the actual link to the file before approving. Also there is a part with some typos, not your fault since they're there from before, but it would be nice to fix them, since you are already re-writing the page!

mikaell commented:
> Why killed? I think it's good, and I think it is fine to use the link to the demo yaml file for reference but I was waiting for you to add the actual link to the file before approving. Also there is a part with some typos, not your fault since they're there from before, but it would be nice to fix them, since you are already re-writing the page!

I've fixed a bunch of typos (let me know if there are more) and added a link to the demo/config-file. The link works clicking around in GitHub, but I don't know how to test it at `https://clinical-genomics.github.io/scout/`. 

Any thing else?

mikaell reviewed (commented):
Now all issues should be fixed 🤠

northwestwitch commented:
> Now all issues should be fixed 🤠
> 
> * Link to example added
> * A lot of Typos
> * Etc
> 
> -There's a change request stopping now which I think is already fixed.

Thanks! I'll re-review ASAP. Still trying to fix those many issues from the Flask major update yesterday!

northwestwitch commented on docs/admin-guide/load-config.md:
```diff
@@ -1,125 +1,93 @@
+
+The format is [Yaml][yaml].
+
+Example configuration files are found here: `<scout root dir>/scout/demo/643594.config.yaml`. [Link](../../scout/demo/643594.config.yaml).
```
The link to this file doesn't work. If you run the preview:
- `mkdocs build`
-` mkdocs serve`

It looks like this:

![image](https://user-images.githubusercontent.com/28093618/118081174-25d67500-b3bb-11eb-967d-b7af56cc8632.png)

dnil replied:
It is perhaps best to link to the repository. AFAIK only the docs get copied to the GitHub IO doc site area on deploy.

mikaell replied:
> It is perhaps best to link to the repository. AFAIK only the docs get copied to the GitHub IO doc site area on deploy.

Good idea! Implemented now: `08c1904`.

northwestwitch reviewed (approved):
Looking good except for the link to the config file that doesn't work. I'm approving but please fix it before merging!

northwestwitch commented on docs/admin-guide/load-config.md:
```diff
@@ -4,7 +4,7 @@ A case and its individuals (or samples, in cancer track) can be uploaded into Sc
 The format is [Yaml][yaml].
 
-Example configuration files are found here: `<scout root dir>/scout/demo/643594.config.yaml`. [Link](../../scout/demo/643594.config.yaml).
+Example configuration files are found here: `<scout root dir>/scout/demo/643594.config.yaml` or on [Github](https://github.com/Clinical-Genomics/scout/blob/update_config_docs-210428/scout/demo/643594.config.yaml).
```
👍🏻
